Browser/assertion_engine.py

```diff
@@ -3,32 +3,83 @@
 import re
 
 AssertionOperator = Enum(
-    "AssertionOperator", "NO_ASSERTION == != contains matches < > <= >="
+    "AssertionOperator",
+    "NO_ASSERTION == should_be != should_not_be < > <= >= *= contains matches ^= $="
 )
 
 
-def verify_assertion(value: Any, operator: AssertionOperator, expected, message=""):
-    if operator.name == "==" and value != expected:
-        raise AssertionError(f"{message} `{value}` should be `{expected}`")
-    if operator.name == "!=" and value == expected:
-        raise AssertionError(f"{message} `{value}` should not be `{expected}`")
-    if operator.name == "<" and value >= expected:
-        raise AssertionError(f"{message} `{value}` should be less than `{expected}`")
-    if operator.name == ">" and value <= expected:
-        raise AssertionError(f"{message} `{value}` should be greater than `{expected}`")
-    if operator.name == "<=" and value > expected:
+def _normalize(operator: str):
+    return operator.lower().replace('_', '').replace(' ', '')
+
+
+def verify_assertion(value: Any,
+                     operator: AssertionOperator,
+                     expected, message=""):
+
+    if isinstance(operator, str):
+        if _normalize(operator) == "contains":
+            operator = AssertionOperator["*="]
+        elif _normalize(operator) == "shouldcontain":
+            operator = AssertionOperator["*="]
+        elif _normalize(operator) == "matches":
+            operator = AssertionOperator["matches"]
+        elif _normalize(operator) == "shouldbe":
+            operator = AssertionOperator["=="]
+        elif _normalize(operator) == "shouldnotbe":
+            operator = AssertionOperator["!="]
+        elif _normalize(operator) == "shouldstartwith":
+            operator = AssertionOperator["^="]
+        elif _normalize(operator) == "shouldendwith":
+            operator = AssertionOperator["$="]
+        else:
+            raise SyntaxError(f"{message}"
+                              f" `{operator}` is no valid assertion operator")
+
+    if operator.name in ["==", "should_be"] and not value == expected:
```

Comment:
Is this `in` check necessary after we normalize and convert to a single possible value above?

Reply:
the in is because we do not have enum aliases and two enums are matching the same behaviour.
These are not strings, but two members of the enum.